### PR TITLE
Added commit number of the build on version info

### LINKED ISSUE #177: The /version should include the commit number of the build.
When I ask for the version, I should get the release number, as well as the build number (the commit ID).

### PR DESCRIPTION
**Pull request by gbuisson**
closes #177

### CODE REVIEW

File: project.clj [hunk 1]
```diff
@@ -63,6 +63,14 @@
...
   :javac-options  ["-proc:none"] ;; remove a warning
+  :filespecs [{:type :fn
+               :fn (fn [p]
+                     {:type :bytes :path "ctia-version.txt"
+                      :bytes (str (:out (clojure.java.shell/sh
+                                         "git" "log" "-n" "1" "--pretty=format:%H "))
+                                  (:out (clojure.java.shell/sh
+                                         "git" "symbolic-ref" "--short" "HEAD")))})}]
+
   :profiles {:dev {:dependencies [[cheshire "5.5.0"]
```
Comment (polygloton) [inline project.clj]: This is cool

File: src/ctia/http/routes/version.clj [hunk 2]
```diff
@@ -1,7 +1,17 @@
...
+            [clojure.string :as st]
+            [ctia.schemas.common :refer [VersionInfo]]
+            [compojure.api.sweet :refer :all]
+            [ring.util.http-response :refer :all]))
+
+(defn current-build []
+  (if-let [built-version (io/resource "ctia-version.txt")]
+    built-version
+    (str (:out (shell/sh "git" "log" "-n" "1" "--pretty=format:%H "))
+         (:out (shell/sh "git" "symbolic-ref" "--short" "HEAD")))))
```
Comment (polygloton) [inline src/ctia/http/routes/version.clj]: Let's memo this, or better yet use an atom explicitly like...

```clojure
(or @version (reset version (current-build)))
```

Comment (gbuisson) [inline src/ctia/http/routes/version.clj]: good idea

Comment (gbuisson) [inline src/ctia/http/routes/version.clj]: or just use a def instead ?

Comment (yogsototh) [inline src/ctia/http/routes/version.clj]: I would give my preference to the atom.

Comment (AlexBaranosky) [inline src/ctia/http/routes/version.clj]: avoiding running code at compile time is a rule thumb to save you much pain. .(that `def` would be running code on load 👎 )

Why not just:

```clj
(def current-build
  (memoize (fn []
             (if-let [built-version (io/resource "ctia-version.txt")]
               built-version
               (str (:out (shell/sh "git" "log" "-n" "1" "--pretty=format:%H "))
                    (:out (shell/sh "git" "symbolic-ref" "--short" "HEAD")))))))
```

Comment (gbuisson) [inline src/ctia/http/routes/version.clj]: agreed ;)

**Pull request merged**